pkg/pipelines/pipelines.go

```diff
@@ -0,0 +1,100 @@
+
+	taskSpecOps := []tb.TaskSpecOp{
+		tb.TaskInputs(tb.InputsResource("workspace", v1alpha1.PipelineResourceTypeGit, tb.ResourceTargetPath("damnworkspace"))),
+		tb.TaskOutputs(tb.OutputsResource("workspace", v1alpha1.PipelineResourceTypeGit)),
+		tb.Step("read-docs-old", "ubuntu", tb.StepCommand("/bin/bash"), tb.StepArgs("-c", "ls -la /workspace/damnworkspace/docs/README.md")),
+		tb.Step("write-new-stuff", "ubuntu", tb.StepCommand("bash"), tb.StepArgs("-c", "ln -s /workspace/damnworkspace /workspace/output/workspace && echo some stuff > /workspace/output/workspace/stuff")),
+	}
+
+	return tb.Task(taskname, namespace, tb.TaskSpec(taskSpecOps...))
+}
+
+func getReadFileTask(taskname string, namespace string) *v1alpha1.Task {
+
+	taskSpecOps := []tb.TaskSpecOp{
+		tb.TaskInputs(tb.InputsResource("workspace", v1alpha1.PipelineResourceTypeGit, tb.ResourceTargetPath("newworkspace"))),
+		tb.Step("read", "ubuntu", tb.StepCommand("/bin/bash"), tb.StepArgs("-c", "cat /workspace/newworkspace/stuff")),
+	}
+
+	return tb.Task(taskname, namespace, tb.TaskSpec(taskSpecOps...))
+}
+
+func getPipeline(pipelineName string, namespace string, createFiletaskName string, readFileTaskName string) *v1alpha1.Pipeline {
+
+	pipelineSpec := []tb.PipelineSpecOp{
+		tb.PipelineDeclaredResource("source-repo", "git"),
+		tb.PipelineTask("first-create-file", createFiletaskName,
+			tb.PipelineTaskInputResource("workspace", "source-repo"),
+			tb.PipelineTaskOutputResource("workspace", "source-repo"),
+		),
+		tb.PipelineTask("then-check", readFileTaskName,
+			tb.PipelineTaskInputResource("workspace", "source-repo", tb.From("first-create-file")),
+		),
+	}
+
+	return tb.Pipeline(pipelineName, namespace, tb.PipelineSpec(pipelineSpec...))
+}
+
+func CreateSamplePiplines(c *client.Clients, namespace string) {
+	var err error
+
+	log.Printf("Creating Git PipelineResource %s", tePipelineGitResourceName)
+
+	_, err = c.PipelineResourceClient.Create(getGitResourceForOutputPipeline(tePipelineGitResourceName, namespace))
+	helper.AssertNoError(err, fmt.Sprintf("Failed to create Pipeline Resource `%s`", tePipelineGitResourceName))
+
+	log.Printf("Creating Task  %s", createFileTaskName)
+
+	_, err = c.TaskClient.Create(getCreateFileTask(createFileTaskName, namespace))
+	helper.AssertNoError(err, fmt.Sprintf("Failed to create Task Resource `%s`", createFileTaskName))
+
+	log.Printf("Creating Task  %s", readFileTaskName)
+
+	_, err = c.TaskClient.Create(getReadFileTask(readFileTaskName, namespace))
+	helper.AssertNoError(err, fmt.Sprintf("Failed to create Task Resource `%s`", readFileTaskName))
+
+	log.Printf("Create Pipeline %s", tePipelineName)
+
+	_, err = c.PipelineClient.Create(getPipeline(tePipelineName, namespace, createFileTaskName, readFileTaskName))
+	helper.AssertNoError(err, fmt.Sprintf("Failed to create pipeline `%s`", tePipelineName))
+
+	log.Println("Created sample pipleine successfully....")
+}
+
+func StartPipeline(path, namespace string) {
```

Comment:
Why do you mix approaches? First you create all resources using API and then start the pipeline using CLI.